### PR TITLE
Add NumPy Argpartition Frontend and Backend

### LINKED ISSUE #9219: argpartition
Add  Argpartition to NumPy Frontend

### PR DESCRIPTION
**Pull request by hossamasaad**
fixes #9219

### CODE REVIEW

**Comment by hossamasaad** [comment]
Argpartition #3607

**Comment by Daniel4078** [comment]
Please, create an issue with title "Argpartition" and then post the link to the issue to the #3607 Todolist issue and this pull request so we can correctly track the status of this task. Thank you. Also, you need to write a test for the function you implemented, check the deep dive document for instructions.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 1]
```diff
@@ -143,6 +143,50 @@ def test_numpy_ndarray_argmax(
...
+    dtype_x_axis,
+    as_variable: pf.AsVariableFlags,
+    native_array: pf.NativeArrayFlags,
+    init_num_positional_args: pf.NumPositionalArgFn,
+    method_num_positional_args: pf.NumPositionalArgMethod,
+    frontend,
+    frontend_method_data,
+):
+    input_dtype, x, axis = dtype_x_axis
+    kth = np.random.randint(low=0, high=max(len(x)-1, 1))
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:167]: you should put the generation of kth in to the "handle_frontend_method" part, and the way you pass it in is also wrong:
TypeError: argpartition() got some positional-only arguments passed as keyword arguments: 'kth'

File: ivy/functional/backends/numpy/searching.py [hunk 2]
```diff
@@ -55,6 +55,32 @@ def argmin(
...
+    if output_dtype is not None:
+        output_dtype = ivy.as_native_dtype(output_dtype)
+        if output_dtype not in (np.int32, np.int64):
+            return ret.astype(dtype=np.int64)
+        else:
+            return ret.astype(dtype=output_dtype)
+    return ret.astype(dtype=np.int64)
+
+    
 argmin.support_native_out = True
```
Comment (Daniel4078) [inline ivy/functional/backends/numpy/searching.py:84]: By the way, you should keep this line right after the argmin function and also add a similar line for argpartition stating that it does not support native out

File: ivy/functional/backends/numpy/searching.py [hunk 3]
```diff
@@ -55,6 +55,32 @@ def argmin(
...
+    *,
+    axis: Optional[Union[int, None]] = -1,
+    kind: Optional[str] = 'introselect',
+    order: Optional[Union[str, list]] = None,
+    output_dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None
+) -> np.ndarray:
+    # Check if the given order one of the field else return pass order as None
+    try:
+        x[order]
+    except:
```
Comment (Daniel4078) [inline ivy/functional/backends/numpy/searching.py:71]: E722 do not use bare 'except'

File: ivy/functional/frontends/numpy/sorting_searching_counting/searching.py [hunk 4]
```diff
@@ -46,6 +46,11 @@ def argmax(
 ):
     return ivy.argmax(a, axis=axis, out=out, keepdims=keepdims)
```
Comment (Daniel4078) [inline ivy/functional/frontends/numpy/sorting_searching_counting/searching.py:48]: E302 expected 2 blank lines, found 1

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py (hunk 1 above)
Comment (Daniel4078) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:167]: E226 missing whitespace around arithmetic operator

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Hello, I found some error for you to fix. Please check the details in the comments below. Thank you,

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py (hunk 1 above)
Comment (hossamasaad) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py]: I added 'kth' to "handle_frontend_method" part using "st.integers()" but I can't set max_value to length of x, if you have an idea how to do that ? @Daniel4078

**Comment by Daniel4078** [comment]
about how to achieve this: set max_value to length of x, you can have a look at this function(https://github.com/unifyai/ivy/blob/8125500b4e1db075afa86553a75e638f6179827e/ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py#L12). It is used in the test right after it to generate value depending on other value generated. You can find other examples in the deep dive document I think. In addition, your functionwrapper seems to introduce some infinite loop to your function: tensorflow: nested_map: maximum recursion depth exceeded in comparison.

**Pull request closed without merging**